eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/FailureMessage.scala

```diff
@@ -89,7 +89,7 @@ object FailureMessageCodecs {
   val NODE = 0x2000
   val UPDATE = 0x1000
 
-  val channelUpdateCodecWithType = meteredLightningMessageCodec.narrow[ChannelUpdate](f => Attempt.successful(f.asInstanceOf[ChannelUpdate]), g => g)
+  val channelUpdateCodecWithType = lightningMessageCodec.narrow[ChannelUpdate](f => Attempt.successful(f.asInstanceOf[ChannelUpdate]), g => g)
```

Comment:
That's not a good fix, the metered version adds metrics and we want to keep that. The problem is that we don't check that `f` is actually a `ChannelUpdate`.

```suggestion
  val channelUpdateCodecWithType = meteredLightningMessageCodec.narrow[ChannelUpdate]({
    case f: ChannelUpdate => Attempt.successful(f)
    case _ => Attempt.failure()
  }, g => g)
```

Reply:
That looks like a good solution too. 

Is metering useful here? it appears this codec parses a `ChannelUpdate` messages that is returned as part of a `FailureMessage`, but a newly received `ChannelUpdate` will still go through `meteredLightningMessageCodec`.

Reply:
I don't really know if metering is useful but I'm assuming it was put there for a reason, so we shouldn't remove it without a good reason. @t-bast what metrics are we interested in? Should this codec be metered?
Anyway I think it's worth fixing the `asInstanceOf[ChannelUpdate]`. There might be some edge case where a tagless `ChannelUpdate` happens to parse as a valid lightning message.
You also need to rebase on master to solve some conflicts.

Reply:
> You also need to rebase on master to solve some conflicts.

Oh thanks; forgot to do that. I just pushed the rebase.

Reply:
> Anyway I think it's worth fixing the `asInstanceOf[ChannelUpdate]`. There might be some edge case where a tagless `ChannelUpdate` happens to parse as a valid lightning message.
> You also need to rebase on master to solve some conflicts.

Your suggested fix is in [364f81c](https://github.com/ACINQ/eclair/pull/2495/commits/364f81ca4bf00e622ccd21973302dd7303c55678). I wanted to do something like that initially, but couldn't figure out the syntax.